kernel/queue.c

```diff
@@ -148,26 +148,21 @@ void k_queue_append_list(struct k_queue *queue, void *head, void *tail)
 {
 	__ASSERT(head && tail, "invalid head or tail");
 
-	unsigned int key = irq_lock();
+	int need_sched = 0;
+	unsigned int was_nonempty, key = irq_lock();
 #if !defined(CONFIG_POLL)
-	struct k_thread *first_thread, *thread;
+	struct k_thread *thread;
 
-	first_thread = _peek_first_pending_thread(&queue->wait_q);
```

Comment:
Maybe get rid of this function now that nothing else calls it?

Reply:
That will come.  There are actually a *lot* of dead functions in ksched.h.  I think the way this is evolving is that I'll end up writing a new header and adding back only what is needed.